crates/ruff/src/rules/pyflakes/fixes.rs

```diff
@@ -69,32 +60,131 @@ pub fn remove_unused_keyword_arguments_from_format_call(
+            conversion_spec,
+            format_spec,
+        } => {
+            let mut field_name = field_name;
+            if let Some(conversion) = conversion_spec {
+                field_name.push_str(&format!("!{conversion}"));
+            }
+            if !format_spec.is_empty() {
+                field_name.push_str(&format!(":{format_spec}"));
+            }
+            format!("{{{field_name}}}")
+        }
+    }
+}
+
+pub fn update_field_types(format_string: &FormatString, min_unused: usize) -> String {
+    format_string
+        .format_parts
+        .iter()
+        .map(|part| match part {
+            FormatPart::Literal(literal) => FormatPart::Literal(literal.to_string()),
+            FormatPart::Field {
+                field_name,
+                conversion_spec,
+                format_spec,
+            } => {
+                let new_field_name = FieldName::parse(field_name).unwrap(); // This should never fail because we parsed it before
+                let mut new_field_name_string = match new_field_name.field_type {
+                    FieldType::Auto => String::new(),
+                    FieldType::Index(i) => (i - min_unused).to_string(),
+                    FieldType::Keyword(kw) => kw,
+                };
+                for field_name_part in &new_field_name.parts {
+                    let field_name_part_string = match field_name_part {
+                        FieldNamePart::Attribute(attribute) => format!(".{attribute}"),
+                        FieldNamePart::Index(i) => format!("[{i}]"),
+                        FieldNamePart::StringIndex(s) => format!("[{s}]"),
+                    };
+                    new_field_name_string.push_str(&field_name_part_string);
+                }
+                let new_format_spec = FormatString::from_str(format_spec).unwrap(); // This should never fail because we parsed it before
+                let new_format_spec_string = update_field_types(&new_format_spec, min_unused);
+                FormatPart::Field {
+                    field_name: new_field_name_string,
+                    conversion_spec: *conversion_spec,
+                    format_spec: new_format_spec_string,
+                }
+            }
+        })
+        .map(unparse_format_part)
+        .collect()
+}
 
-    body.value = Expression::Call(Box::new(new_call));
+/// Generate a [`Fix`] to remove unused keyword arguments from format call.
+pub fn remove_unused_positional_arguments_from_format_call(
+    unused_arguments: &[usize],
+    location: Range,
+    locator: &Locator,
+    stylist: &Stylist,
+    format_string: &FormatString,
+) -> Result<Fix> {
+    let module_text = locator.slice(location);
+    let mut tree = match_expression(module_text)?;
+    let mut call = match_call(&mut tree)?;
+
+    call.args = call
+        .args
+        .iter()
+        .enumerate()
+        .filter_map(|(i, e)| (!unused_arguments.contains(&i)).then_some(e.clone()))
+        .collect();
```

Comment:
Nit: Can we use `retain` or `std::mem::take` with `into_iter`?

Reply:
Done.